app/screens/emoji_picker/picker/emoji_category_bar/index.tsx

```diff
@@ -45,22 +94,81 @@ const EmojiCategoryBar = ({onSelect}: Props) => {
+            currentIndex={currentIndex}
+            key={icon.key}
+            icon={icon.icon}
+            index={index}
+            scrollToIndex={scrollToIndex}
+            theme={theme}
+        />
+    ));
+
     return (
-        <View
-            style={styles.container}
-            testID='emoji_picker.category_bar'
-        >
-            {icons.map((icon, index) => (
-                <EmojiCategoryBarIcon
-                    currentIndex={currentIndex}
-                    key={icon.key}
-                    icon={icon.icon}
-                    index={index}
-                    scrollToIndex={scrollToIndex}
-                    theme={theme}
-                />
-            ))}
-        </View>
+        <>
+            {isEmojiPicker &&
+                <View
+                    style={styles.container}
+                    testID='emoji_picker.category_bar'
+                >
+                    {handleToggleEmojiPicker && deleteCharFromCurrentCursorPosition &&
+                    <View style={styles.categoryBar}>
+                        <View style={styles.keyboardControls}>
+                            <Pressable
+                                onPress={() => handleToggleEmojiPicker()}
+                                style={styles.actionIcons}
+                            >
+                                <CompassIcon
+                                    name={'keyboard-outline'}
+                                    size={20}
+                                    color={changeOpacity(theme.centerChannelColor, 0.56)}
+                                />
+                            </Pressable>
+                            <Divider
+                                orientation='vertical'
+                                color={changeOpacity(theme.centerChannelColor, 0.08)}
+                                width={1}
+                            />
+                        </View>
+                        <View style={{...styles.category, flex: isTablet ? 0 : 1}}>
+                            {!isTablet &&
+                            <ScrollView
+                                horizontal={true}
+                                showsHorizontalScrollIndicator={false}
+                                contentContainerStyle={styles.scrollView}
+                            >
+                                {iconCatergories}
+                            </ScrollView>
+                            }
+                            {isTablet && iconCatergories}
+                        </View>
+                        <View style={styles.keyboardControls}>
+                            <Divider
+                                orientation='vertical'
+                                color={changeOpacity(theme.centerChannelColor, 0.08)}
+                                width={1}
+                            />
+                            <Pressable
+                                onPress={() => deleteCharFromCurrentCursorPosition()}
```

Comment:
Unless you are making sure no arguments are passed to `deleteCharFromCurrentCursorPosition`, the following can be done:

```suggestion
                                onPress={deleteCharFromCurrentCursorPosition}
```

If it is important to avoid the argument, use a `useCallback` to avoid passing a new function on every render.